src/components/ButtonPanel.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { v4 as genKey } from 'uuid';
+import Button from './Button';
+
+const renderGroup = group => (
+  group.map(item => <Button name={item} key={genKey()} />)
+);
+
+const ButtonPanel = () => {
+  const groups = {
+    group1: ['AC', '+/-', '%', '÷'],
+    group2: ['7', '8', '9', 'X'],
+    group3: ['4', '5', '6', '-'],
+    group4: ['1', '2', '3', '+'],
+    group5: ['0', '.', '='],
+  };
+
+  return (
+    <div className="button-panel">
+      {Object.entries(groups).map(group => (
+        <div className="button-row" key={genKey()}>
+          {' '}
+          {renderGroup(group[1])}
+          {' '}
+        </div>
+      ))}
```

Comment:
Nice! ⭐ 

Is the white space really necessary? `{' '}`.

OPTIONAL
---

Is the key going to be generated on every render? Remember that Every time a component's key changes _[React will create a new component instance rather than update the current one](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key:~:text=React%20will%20create%20a%20new%20component%20instance%20rather%20than%20update%20the%20current%20one.)._

You also have a unique stable value already that you could use so React can keep track if the component has changed:

```js
{Object.entries(groups).map(([groupName, group]) => (
  <div className="button-row" key={groupName}>
    {group.map(item => <Button name={item} key={item} />)}
  </div>
))}
```
---

It's genuinely nice how you solve this, and it is also nice that you're using a package to generate the keys.

Just be mindful about performance issues if a new key is generated every time.

I'm not sure if that's going to be the case here and anyhow is not going to be a performance issue in an small project like this one, but I'm going to keep track of your project to see how it behaves and learn from it😁.

Reply:
Hi Mauricio, thanks for your feedback!

That white space might have been product of a linter autofix. I just removed it.
And yes, I agree that's a better approach for the key values on the rows. My previous implementation used an array instead of an object, so I didn't have the group names available on each iteration. Now that I've made the change to an object, I can use the names for key values. I'm implementing this right now.

Thanks again for your tips!